utoipa-gen/src/schema.rs

```diff
@@ -100,7 +84,12 @@ impl<'a> ComponentPart<'a> {
             );
         };
 
-        let mut generic_component_type = ComponentPart::convert(&segment.ident, segment);
+        let path = TypePath {
+            qself: None,
+            path: syn::Path::from(segment.clone()),
+        };
+
+        let mut generic_component_type = ComponentPart::convert(Cow::Owned(path));
```

Comment:
Hmm.. interesting approach :thinking:

Reply:
Do you think there is a better way to do it?